src/main/java/raythx98/duke/task/TaskList.java

```diff
@@ -0,0 +1,71 @@
+package raythx98.duke.task;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import raythx98.duke.exception.InvalidIndexException;
+
+/**
+ * Deals with manipulation on the list of tasks
+ */
+public class TaskList {
+
+    private List<Task> taskList = new ArrayList<>();
+
+    /**
+     * Removes a task from List of tasks based on the index.
+     *
+     * @param idx Index of task to be removed.
+     * @return String which represents the task removed.
+     */
+    public String removeTask (int idx) {
+        Task task = taskList.get(idx);
+        taskList.remove(idx);
+        return task.toString();
+    }
+
+    public Task getTask(int idx) {
+        return taskList.get(idx);
+    }
+
+    public int getSize() {
+        return taskList.size();
+    }
+
+    public void addTask(Task task) {
+        taskList.add(task);
+    }
+
+    /**
+     * Returns a list of all tasks at the moment.
+     *
+     * @return String which lists all tasks at the moment.
+     */
+    public String findTask(String str) throws InvalidIndexException {
+        TaskList match = new TaskList();
+        for (Task i: taskList) {
+            String taskDescription = i.getDescription().toLowerCase();
+            String toCompare = str.toLowerCase();
+            if (taskDescription.contains(toCompare)) {
+                match.addTask(i);
+            }
+        }
+        if (match.getSize() == 0) {
+            throw new InvalidIndexException("No such tasks leh...");
+        }
+        return "Got it, here yur task bij\n" + match.toString();
+    }
+
+    public String toString() {
+        String string = "";
```

Comment:
Should this String have a more descriptive name?

Reply:
Agreed, will resolve